src/fields/fft_poisson_solver/fft/WrapRocDST.cpp

```diff
@@ -236,107 +221,228 @@ namespace AnyDST
     {
         HIPACE_PROFILE("AnyDST::CreatePlan()");
         DSTplan dst_plan;
-        const int nx = real_size[0];
-        const int ny = real_size[1];
-        int dim = 2;
-
-        // Allocate expanded_position_array Real of size (2*nx+2, 2*ny+2)
-        // Allocate expanded_fourier_array Complex of size (nx+2, 2*ny+2)
-        amrex::Box expanded_position_box {{0, 0, 0}, {2*nx+1, 2*ny+1, 0}};
-        amrex::Box expanded_fourier_box {{0, 0, 0}, {nx+1, 2*ny+1, 0}};
-        // shift box to match rest of fields
-        expanded_position_box += position_array->box().smallEnd();
-        expanded_fourier_box += fourier_array->box().smallEnd();
-        dst_plan.m_expanded_position_array =
-            std::make_unique<amrex::FArrayBox>(
-                expanded_position_box, 1);
-        dst_plan.m_expanded_fourier_array =
-            std::make_unique<amrex::BaseFab<amrex::GpuComplex<amrex::Real>>>(
-                expanded_fourier_box, 1);
-
-        // setting the initial values to 0
-        // we don't set the expanded Fourier array, because it will be initialized by the FFT
-        dst_plan.m_expanded_position_array->setVal<amrex::RunOn::Device>(0.,
-            dst_plan.m_expanded_position_array->box(), 0,
-            dst_plan.m_expanded_position_array->nComp());
-
-        // check for type of expanded size, should be const size_t *
-        const amrex::IntVect& expanded_size = expanded_position_box.length();
-        const std::size_t lengths[] = {AMREX_D_DECL(std::size_t(expanded_size[0]),
-                                                    std::size_t(expanded_size[1]),
-                                                    std::size_t(expanded_size[2]))};
+
+        amrex::ParmParse pp("hipace");
+        dst_plan.use_small_dst = (std::max(real_size[0], real_size[1]) >= 511);
+        queryWithParser(pp, "use_small_dst", dst_plan.use_small_dst);
+
```

Comment:
```suggestion
        // Bitwise operation to determine whether nx or ny are powers of 2
        const bool nx_power_of_2 = (real_size[0] & (real_size[0] - 1)) == 0
        const bool ny_power_of_2 = (real_size[1] & (real_size[1] - 1)) == 0
        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
            (nx_power_of_2 && ny_power_of_2) || (dst_plan.use_small_dst == true),
            "big DST only works for transverse domain sizes that are powers of 2. Otherwise, use hipace.use_small_dst = 1"
            );
```
I believe we have a bug with big DST + 2^n-1 points on AMD. Could you assert that the user doesn't use that?

Reply:
Please don't commit that for now, it'll affect some tests we're currently running. After these tests, we may revisit this assert.

Reply:
Let's keep that for a follow-up PR, and maybe do more tests.